src/traccuracy/metrics/_base.py

```diff
@@ -6,32 +8,22 @@
 
 
 class Metric(ABC):
+    """The base class for Metrics
+
+    Data should be passed directly into the compute method
+    Kwargs should be specified in the constructor
+    """
+
     # Mapping criteria
     needs_one_to_one = False
     supports_one_to_many = False
     supports_many_to_one = False
     supports_many_to_many = False
 
-    def __init__(self, matched_data: "Matched"):
-        """Add Matched class which takes TrackingData objects for gt and pred,and computes matching
-
-        Each current matching method will be a subclass of Matched e.g. CTCMatched or IOUMatched.
-        The Matched objects will store both gt and pred data, as well as the mapping,
-        and any additional private attributes that may be needed/used e.g. detection matrices.
-        Metric subclasses will take keyword arguments to set the weights of various error counts.
-
-        Args:
-            matched_data (Matched): Matched object for set of GT and Pred data
-        """
-        self.data = matched_data
-        self.results = self.compute()
```

Comment:
I like this change as well - it makes sense that the Metric wouldn't store the results if it doesn't store the data.